app/forms/personal_details_form.rb

```diff
@@ -159,4 +160,27 @@ def reset_blank_nationalities
 
     params.merge!(raw_values)
   end
+
+  def raw_values_nationalities_array
+    [other_nationality1_raw, other_nationality2_raw, other_nationality3_raw]
+  end
+
+  def check_raw_values
+    # check the freetext responses of the user, and if
+    # they are valid responses, we use them by populating new attributes.
+    # This was to fix a bug where valid responses were not being used
+    # as they were not selected from the dropdown list
+
+    raw_values_nationalities_array.each_with_index do |raw_value, index|
+      next unless find_nationality(raw_value)
+
+      public_send("other_nationality#{index + 1}=", raw_value)
+      new_attributes[:"other_nationality#{index + 1}"] << raw_value
+      nationality_ids[index] = find_nationality(raw_value).id
+    end
+  end
+
+  def find_nationality(raw_value)
+    Nationality.find_by(name: raw_value)
```

Comment:
Maybe this:
```suggestion
    Nationality.find_by(name: raw_value.downcase)
```
Then we get any other CaSe issUEss for FreE!

Reply:
All done 👍